what4/src/What4/Protocol/SMTWriter.hs

```diff
@@ -1604,17 +1604,20 @@ bvIntTerm w x = sumExpr ((\i -> digit (i-1)) <$> [1..natValue w])
                      (fromInteger (2^d))
                      0
 
-sbvIntTerm :: SupportTermOps v
+sbvIntTerm :: (SupportTermOps v, 1 <= w)
            => NatRepr w
            -> v
            -> v
-sbvIntTerm w0 x0 = sumExpr (signed_offset : go w0 x0 (natValue w0 - 2))
+sbvIntTerm w0 x0 = sumExpr (signed_offset : go w0 x0 (intValue w0 - 2))
```

Comment:
This took a little bit of staring to realize the main problem is this line and not the one with the explaining comment below.  This works, although I still worry that the super-degenerate case of a length-0 bitvector would trigger an underflow exception in `signed_offset`.   An alternative could be:
```
sbvIntTerm w0 _ | natValue w0 < 2 = sumExpr [0]
sbvIntTerm w0 x0 = ... original unchanged and dropping the `otherwise` clause below ...
```
although I'm not sure about the `sumExpr` arg for the length-1 bitvector case.

Reply:
Although I had seen the constraint you added on `w`, I was worried that it could be called with a `v` that was a different size than was represented by `w`.  However, some light checking indicates that this is unlikely and would be very difficult to do (if even possible), so as a result I'm less concerned about this issue.  Adjusting the comment location would be appreciated though.

Reply:
The longer that I stare at the implementation of `sbvIntTerm`, the more I realize how unnecessarily complicated it is. It's doing basically the same thing as `bvIntTerm`, but adding an additional summand to make it negative if need be. That's it!

I've pushed an additional commit that refactors the code to make the similarities between `bvIntTerm` and `sbvIntTerm` more obvious. As a bonus, this makes it much clearer where the potential underflow issue would be (in the newly christened `powersOfTwo` helper functions). Let me know if you like this version better.